source/blog/2017-06-22-introducing-buildah.html.md

```diff
@@ -0,0 +1,258 @@
+#
+```
+
+We don't need any other container runtime, either.  This is very useful because it makes Buildah easily embeddable in automation scripts and inside other container-building tools, since neither a server process nor root access are required.
+
+## Manipulating the Image
+
+In the initial buildah `from` command, I saved the container name in the `$container` variable, and I can now use that variable to manipulate the container.   First, let's mount the container on a local mount point, saving the location of the mount point in the `$mountpoint` variable and then we'll add a file from our local system into our container.  To do this, we use:
+
+```
+# mountpoint=$(buildah mount $container)
+```
+
+Now that we've the mountpoint in hand, let's play around a bit with it:
+
+```
+# ls $mountpoint
+bin   dev  home  lib64  mnt  proc  run   srv  tmp  var
+boot  etc  lib   media  opt  root  sbin  sys  usr
+# cat /tmp/moveit.txt
+Move this file to the Buildah container's /tmp directory.
+# ls $mountpoint/tmp
+anaconda-post.log  ks-script-64xpp1a9  ks-script-sw0bp7j4
+# cp /tmp/moveit.txt $mountpoint/tmp
+# ls $mountpoint/tmp
+anaconda-post.log  ks-script-64xpp1a9  ks-script-sw0bp7j4  moveit.txt
+# cat $mountpoint/tmp/moveit.txt
+Move this file to the Buildah container's /tmp directory.
+```
+
+So in under 30 seconds we were able to pull down an image from a registry, create a corresponding container and then manipulate that container.  We now have a container that's unique to our needs, so let's save an image of it.  The `buildah commit` command creates a new image which will let us replicate the container later as we need to.  After we create and save the image we'll remove the first container that we created and then create a second container from our newly created image.  With luck our `moveit.txt` file will be in the new container's /tmp directory!
+
+Let's create an image based upon our container.
+
+```
+# buildah commit $container containers-storage:first-new-image
+# buildah images
+IMAGE ID        IMAGE NAME
+dc3a60619440 docker.io/library/fedora:latest
+54e012885dcf docker.io/library/first-new-image:latest
+```
+
+Now let's clean up our containers leaving only our images behind.
+
+```
+# buildah umount "$container"
+# buildah rm "$container"
+ae93de45778a5f3905c52de9378f14d54c2617e4336766efde6311596541e98c
+# buildah containers  # No more containers
+```
+
+If we want to create a new container using the saved image, we just execute the following:
+
+```
+# container2=$(buildah from first-new-image)
+# buildah containers
+CONTAINER ID IMAGE ID     IMAGE NAME      CONTAINER NAME
+757b5b595f0a 54e012885dcf first-new-image first-new-image-working-container
+```
+
+Let's mount the container on a local mount point:
+
+```
+# mountpoint2=$(buildah mount $container2)
+```
+
+Now check to see if our `moveit.txt` file is in the new container that we created from the image.
+
+```
+# ls $mountpoint2/tmp  # Let's see if our file is there
```

Comment:
please remove "# Let's see if our file is there", it's redundant and thought I'd removed it this morning.